test/hotspot/jtreg/runtime/jni/daemonDestroy/exedaemonDestroy.c

```diff
@@ -102,4 +113,24 @@ int main(int argc, char *argv[]) {
     exit(1);
   }
   puts("Test: DestroyJavaVM returned");
+  return 0;
+}
+
+int main(int argc, char *argv[]) {
+   args_list args;
+   args.argc = argc;
+   args.argv = argv;
+#ifdef AIX
+   size_t adjusted_stack_size = 1024*1024;
+   pthread_t id;
+   pthread_attr_t attr;
+   pthread_attr_init(&attr);
+   pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_JOINABLE);
+   pthread_attr_setguardsize(&attr, 0);
+   pthread_attr_setstacksize(&attr, adjusted_stack_size);
```

Comment:
Do we really need this? All other platforms seem to live fine with the default stack size.
OTOH primordial thread stack default size and pthread default stack size may differ, maybe its better to be explicit.

Reply:
When NULL is passed as argument `pthread_create(&id, NULL, run, (void *)&args)` , the test is failing with an exit value 139. To solve that i have included the functions `pthread_attr_setdetachstate` , `pthread_attr_setguardsize` and `pthread_attr_setstacksize` to set attributes for the thread. 
Commenting `pthread_attr_setdetachstate` and `pthread_attr_setguardsize` the test has passed. I will update it here.

Thank you @tstuefe